woocommerce/woocommerce
# PoC: Integrate Block Binding API

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This PR uses the `core/paragraph` block in the `woocommerce/product-text-area-field` block.
Relevant changes:

### Replace RichText component with block template

It defines a template for the text-area block, creating a `core/paragraph ` instance as part of it.

### Connect the `core/paragraph` block instance with the `entity prop`

The entity prop [sets in the `property` attribute of the `woocommerce/product-text-area-field` block](https://github.com/woocommerce/woocommerce/blob/109bb828f133a514d7af4f472c9352bce620c8de/plugins/woocommerce/src/Internal/Features/ProductBlockEditor/ProductTemplates/SimpleProductTemplate.php#L234). 

Depending on the Binding Block API availability, it registers the binding-source handler or extends the `core/paragraph` block via the registerbLockType filter to implement a temporary fallback process.





Closes # .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Since the Block Cinding API is not available yet, the first thing to test is everything works as expected

1. Activate the new Product Editor
2. Go to the General tab.
3. Identify Basic details section
4. Focus on the Summary field block
5. Confirm it works as usual
  * Edit the content
  * Save the product
  * Hard refresh
Additionally, you can take a look a the requests that the client performs when you save the product:

https://github.com/woocommerce/woocommerce/assets/77539/4674c6d9-227e-41f2-a474-a1683a65b443

To test the implementation with the Block Binding API you need the following:

* Use WordPress cover version 6.5
* Use the Gutenberg development version. I've created a branch with the required changes to be able to use the API here. https://github.com/WordPress/gutenberg/pull/58698
* Build the plugin
* Activate the plugin
* Check that it works as usual 

https://github.com/woocommerce/woocommerce/assets/77539/6a6a26a6-48b0-4a76-a640-6e67e3384ad5

Additionally, you check the new `metadata` attribute of the `core/paragraph` block where it binds the instance with the entity prop:

https://github.com/woocommerce/woocommerce/assets/77539/05f19738-b313-4855-8eff-02abd99004d8




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
